manifests/server/rsync_shares.pp

```diff
@@ -1,119 +1,128 @@
 ){
-  include 'rsync::server::global'
+
+  validate_absolute_path($rsync_base)
+  validate_bool($use_stunnel)
+  validate_net_list($hosts_allow)
+
+  include '::rsync::server::global'
+
+  $_rsync_subdir = "${facts['operatingsystem']}/${facts['operatingsystemmajrelease']}"
 
   if $use_stunnel {
-    $l_hosts_allow = '127.0.0.1'
+    $_hosts_allow = ['127.0.0.1']
   }
   else {
-    $l_hosts_allow = $hosts_allow
-  }
-  rsync::server::section { 'default':
-    comment     => 'The default file path',
-    path        => "${rsync_base}/default",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'openldap_server':
-    auth_users  => ['openldap_rsync'],
-    comment     => 'Configuration for OpenLDAP',
-    path        => "${rsync_base}/openldap/server",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'bind_dns_default':
-    auth_users  => ['bind_dns_default_rsync'],
-    comment     => 'Default DNS configurations for named',
-    path        => "${rsync_base}/bind_dns/default",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'apache':
-    auth_users     => ['apache_rsync'],
-    comment        => 'Apache configurations',
-    path           => "${rsync_base}/apache",
-    hosts_allow    => '127.0.0.1',
-    outgoing_chmod => 'o-rwx'
-  }
-  rsync::server::section { 'tftpboot':
-    auth_users  => ['tftpboot_rsync'],
-    comment     => 'Tftpboot server configurations',
-    path        => "${rsync_base}/tftpboot",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'mcafee':
-    comment     => 'McAfee DAT files',
-    path        => "${rsync_base}/mcafee",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'clamav':
-    comment     => 'ClamAV Virus Database Updates',
-    path        => "${rsync_base}/clamav",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'dhcpd':
-    auth_users  => ['dhcpd_rsync'],
-    comment     => 'DHCP Configurations',
-    path        => "${rsync_base}/dhcpd",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'snmp':
-    comment     => 'SNMP MIBs and Modules',
-    path        => "${rsync_base}/snmp",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'freeradius':
-    auth_users  => ['freeradius_systems'],
-    comment     => 'Freeradius configuration files',
-    path        => "${rsync_base}/freeradius",
-    hosts_allow => $l_hosts_allow
-  }
-  rsync::server::section { 'jenkins_plugins':
-    comment     => 'Jenkins Configuration',
-    path        => "${rsync_base}/jenkins_plugins",
-    hosts_allow => $l_hosts_allow,
+    $_hosts_allow = $hosts_allow
   }
 
-  validate_absolute_path($rsync_base)
-  validate_bool($use_stunnel)
-  validate_net_list($hosts_allow)
+  if $facts['simp_rsync_environments'] and !empty($facts['simp_rsync_environments']) {
+    $facts['simp_rsync_environments'].each |String $_env| {
+
```

Comment:
@lnemsick-simp I guess that we could allow them to pass in an array. If the directories don't exist it'll be complete garbage though. I'll change this to give them the option.